File: javascript/socketd/src/socketd/transport/client/ClientHandshakeResult.ts
import type {ChannelInternal} from "../core/Channel";

/**
 * 客户端握手结果
 *
 * @author noear
 * @since 2.0
 */
export class ClientHandshakeResult {
    private _channel: ChannelInternal | null;
    private _throwable: any;

    constructor(channel: ChannelInternal | null, throwable: any) {
        this._channel = channel;
        this._throwable = throwable;
    }


    getChannel(): ChannelInternal | null {
        return this._channel;
    }

    getThrowable(): any {
        return this._throwable;
    }
}

File: python/socketd/socketd/transport/client/ClientHandshakeResult.py

from socketd.transport.core.ChannelInternal import ChannelInternal


class ClientHandshakeResult:

    def __init__(self, __channel: ChannelInternal, __throwable: Exception | None):
        self._channel = __channel
        self._throwable = __throwable

    def get_channel(self):
        return self._channel

    def get_throwable(self):
        return self._throwable
